### PR TITLE
Update Rust to version 1.69

### PR DESCRIPTION
**Pull request by jvff**
# Motivation

Clippy has a `int_to_bool_with_if` lint that is marked as `pedantic` in its latest versions. Since we ran into that lint in #691, the cleanest solution is to update the Rust toolchain to use the new Clippy version.

# Solution

Update Rust to the current latest version: 1.69.0. The new Clippy version printed some new lints, so they were also applied in this PR.

### CODE REVIEW

File: linera-chain/src/chain.rs [hunk 1]
```diff
@@ -184,7 +184,7 @@ where
                 event.is_none(),
                 ChainError::MissingCrossChainUpdate {
                     chain_id,
-                    origin,
+                    origin: origin.into(),
```
Comment (ma2bd) [inline linera-chain/src/chain.rs:187]: Isn't `*origin` shorter / more precise?

File: linera-chain/src/manager.rs [hunk 2]
```diff
@@ -20,9 +20,10 @@ use std::collections::HashMap;
 use tracing::error;
 
 /// How to produce new blocks.
-#[derive(Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
 pub enum ChainManager {
     /// The chain is not active. (No blocks can be created)
+    #[default]
```
Comment (ma2bd) [inline linera-chain/src/manager.rs:26]: nice!

File: linera-core/src/node.rs [hunk 3]
```diff
@@ -203,7 +203,7 @@ impl From<ChainError> for NodeError {
                 height,
             } => Self::MissingCrossChainUpdate {
                 chain_id,
-                origin,
+                origin: *origin,
```
Comment (ma2bd) [inline linera-core/src/node.rs:206]: here we go :)

**Review by ma2bd — APPROVED** [review]
Thanks for the upgrade!

File: linera-chain/src/chain.rs (hunk 1 above)
Comment (jvff) [reply linera-chain/src/chain.rs:187]: No, this one is different from the one in `linera-core/src/node.rs`. This converts from `Origin` to `Box<Origin>`, while the other from `Box<Origin>` to `Origin`.

**Pull request merged**